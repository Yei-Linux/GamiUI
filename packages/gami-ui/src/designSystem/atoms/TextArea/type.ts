import {
  TComponentEvents,
  TDesignTokens,
  TDesignUI,
} from 'core/domain/types/design-system'
import { TOnChangeFormItem } from '../Input/type'
import { IComponentsVariant } from 'core/domain/interfaces/IComponentsVariant'

interface AutoSize {
  minRows: number
  maxRows: number
}

export type TTextArea = {
  /**
   * Placeholder TextArea to show
   */
  placeholder?: string
  /**
   * Identifier TextArea
   */
  name?: string
  /**
   * Value TextArea
   */
  value?: any
  /**
   * Cols TextArea
   */
  cols?: number
  /**
   * Rows TextArea
   */
  rows?: number
  /**
   * Autosize TextArea
   */
  autoSize?: AutoSize
  /**
   * Function to detect changes
   */
  onChangeFormItem?: TOnChangeFormItem
}

export type TTextAreaComponent = TTextArea &
  TDesignTokens &
  TDesignUI &
  TComponentEvents<any, any> &
  IComponentsVariant
