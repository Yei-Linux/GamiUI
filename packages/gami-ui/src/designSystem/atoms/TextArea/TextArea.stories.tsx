import React from 'react'
import { Meta } from '@storybook/react'

import TextArea from '.'

export default {
  title: 'Atoms/TextArea',
  component: TextArea,
  args: {},
  argTypes: {
    placeholder: { control: 'text' },
  },
} as Meta

export const BasicTextArea = (args: any) => <TextArea {...args} />
BasicTextArea.args = {
  placeholder: 'Type your text',
  width: 'full',
}
