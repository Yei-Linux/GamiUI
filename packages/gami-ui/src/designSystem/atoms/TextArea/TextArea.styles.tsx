import styled from '@emotion/styled'
import {
  BorderType,
  FontWeightType,
  HeightType,
  ShadowType,
  TextAlignType,
  WidthType,
} from 'core/domain/types'
import { mixinInput } from 'styles/mixins/input'
import { spacing } from 'styles/tokens'
import { setGenericPropStyles } from 'styles/utilities/genericPropStyles'

export const TextArea = styled.textarea<{
  $border?: BorderType
  $shadow?: ShadowType
  $width?: WidthType
  $heigth?: HeightType
  $textAlign?: TextAlignType
  $fontWeight?: FontWeightType
}>`
  height: 69px;
  min-height: 75px;
  max-height: 225px;

  margin: ${spacing.margin.none};
  padding: ${spacing.padding.md};

  ${mixinInput('light')};

  ${({ $border, $shadow, $fontWeight, $width, $textAlign, $heigth }) =>
    setGenericPropStyles({
      border: $border || 'sm',
      shadow: $shadow || 'sm',
      width: $width || 'auto',
      heigth: $heigth || 'auto',
      fontWeight: $fontWeight,
      textAlign: $textAlign,
    })};
`
