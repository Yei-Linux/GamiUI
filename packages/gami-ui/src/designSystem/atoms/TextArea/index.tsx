export { default } from './TextArea'
export { options } from './constants'
export { argTypes } from './storyConfig'
