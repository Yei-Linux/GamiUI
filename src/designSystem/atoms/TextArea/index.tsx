export { default } from "./TextArea";
export { options } from "./constants";
