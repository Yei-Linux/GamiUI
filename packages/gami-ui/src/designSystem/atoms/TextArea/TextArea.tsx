import React from 'react'
import { getGenericPropStyles } from 'styles/utilities/genericPropStyles'
import { IGeneralProps } from '../../../core/domain/interfaces/IGeneralProps'
import { TOnChangeFormItem } from '../Input/Input'
import * as S from './TextArea.styles'

interface AutoSize {
  minRows: number
  maxRows: number
}

export interface ITextArea extends IGeneralProps {
  /**
   * Placeholder TextArea to show
   */
  placeholder?: string
  /**
   * Identifier TextArea
   */
  name?: string
  /**
   * Value TextArea
   */
  value?: any
  /**
   * Cols TextArea
   */
  cols?: number
  /**
   * Rows TextArea
   */
  rows?: number
  /**
   * Autosize TextArea
   */
  autoSize?: AutoSize
  /**
   * Function to detect changes
   */
  onChangeFormItem?: TOnChangeFormItem
}

const TextArea = ({
  onChangeFormItem,
  name = 'gamification',
  placeholder = '',
  cols = 5,
  rows = 5,
  ...genericsProps
}: ITextArea) => {
  const handleChangeOnTextArea = (
    e: React.ChangeEvent<HTMLTextAreaElement>
  ) => {
    e.preventDefault()
    onChangeFormItem?.(e.target.value)
  }

  return (
    <S.TextArea
      name={name}
      placeholder={placeholder}
      cols={cols}
      rows={rows}
      onChange={handleChangeOnTextArea}
      {...getGenericPropStyles(genericsProps)}
    />
  )
}

export default TextArea
