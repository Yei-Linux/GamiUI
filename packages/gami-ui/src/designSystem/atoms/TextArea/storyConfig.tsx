import { IStoryMainConfig } from 'core/domain/interfaces/IStorybook'
import {
  getInheritGlobalStylesStories,
  getStoryConfigStructure,
  madegenericPropsControl,
} from 'core/helpers/storybook.helper'
import TextArea from '.'

const genericArgTypes = madegenericPropsControl([])

const storyArgTypes = {
  placeholder: {
    control: 'text',
    description: 'placeholder to Show',
    table: {
      type: { summary: 'string' },
    },
  },
  name: {
    control: 'text',
    description: 'name to Show',
    table: {
      type: { summary: 'string' },
    },
  },
  cols: {
    control: 'number',
    description: 'cols to Show',
    table: {
      type: { summary: 'number' },
    },
  },
  rows: {
    control: 'number',
    description: 'cols to Show',
    table: {
      type: { summary: 'number' },
    },
  },
  ...genericArgTypes,
}

const docArgTypes = {
  onChangeFormItem: {
    control: 'object',
    description: 'onChangeFormItem Prop',
    table: {
      type: { summary: '() => void' },
    },
  },
}

const argTypes = { ...storyArgTypes, ...docArgTypes }

const mainConfig: IStoryMainConfig = {
  title: 'Atoms/TextArea ðŸŸ¢',
  component: TextArea,
  args: {},
  argTypes: storyArgTypes,
}

const storiesInheritGlobalStyles = getInheritGlobalStylesStories({
  rounded: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  shadow: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  width: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  height: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  size: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  fontWeight: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  textAlign: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
  },
  margin: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px', 'xs:0 0 1rem 0'],
  },
  padding: {
    args: {
      placeholder: 'Type your text',
      width: 'full',
    },
    examples: ['0px', '2px', '4px', '6px', '8px', '10px', 'xs:0 0 1rem 0'],
  },
})

const storiesComponent = [
  {
    storyName: 'WithText ðŸ™‚',
    self: {
      args: { placeholder: 'Type your text', width: 'full' },
      variants: {
        examples: [
          {
            label: 'Basic',
            value: 'This is a TextArea',
          },
        ],
        field: 'text',
      },
    },
  },
]

const storyConfig = getStoryConfigStructure({
  mainConfig,
  storiesInheritGlobalStyles,
  storiesComponent,
  component: TextArea,
})

export { storyConfig, argTypes }
