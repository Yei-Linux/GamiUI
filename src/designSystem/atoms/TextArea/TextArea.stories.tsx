import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import TextArea from ".";

const Template = getTemplate(TextArea);

export default {
  title: "Atoms/TextArea",
  component: TextArea,
  args: { ...TextArea.defaultProps },
  argTypes: {
    placeholder: { control: "text" },
    ...stylesControl,
  },
} as Meta;

export const BasicTextArea = Template.bind({});
BasicTextArea.args = {
  placeholder: "Type your text",
  width: "NORMAL",
  heigth: "SMALL",
  shadow: "",
  border: "MEDIUM"
};