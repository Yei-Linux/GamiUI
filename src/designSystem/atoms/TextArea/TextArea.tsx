import React from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { TextAreaWrapper } from "./TextArea.styles";

interface AutoSize {
  minRows: number;
  maxRows: number;
}

export interface ITextArea extends IGeneralProps {
  /**
   * Placeholder TextArea to show
   */
  placeholder?: string;
  /**
   * Identifier TextArea
   */
  name?: string;
  /**
   * Value TextArea
   */
  value?: any;
  /**
   * Cols TextArea
   */
  cols?: number;
  /**
   * Rows TextArea
   */
  rows?: number;
  /**
   * Autosize TextArea
   */
  autoSize?: AutoSize;
  /**
   * Function to detect changes
   */
  onChangeFormItem?: any;
}

const TextArea = ({ autoSize, onChangeFormItem, ...args }: ITextArea) => {
  return (
    <TextAreaWrapper
      {...args}
      onChange={(e: any) => {
        onChangeFormItem(e.target.value);
        e.preventDefault();
      }}
    />
  );
};

TextArea.defaultProps = {
  name: "gamification",
  placeholder: "",
  cols: 5,
  rows: 5,
  width: "NORMAL",
  heigth: "SMALL",
  border: "MEDIUM",
};

export default TextArea;
