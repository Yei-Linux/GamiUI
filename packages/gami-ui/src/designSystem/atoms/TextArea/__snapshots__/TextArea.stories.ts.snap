// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/TextArea Basic Text Area 1`] = `
<textarea
  className="css-1iivwc3-TextAreaWrapper eieaaae0"
  cols={5}
  fontWeight="BOLD"
  name="gamification"
  onChange={[Function]}
  placeholder="Type your text"
  rows={5}
  width="NORMAL"
/>
`;
